shardingsphere-kernel/shardingsphere-single-table/shardingsphere-single-table-core/src/test/java/org/apache/shardingsphere/singletable/route/validator/ddl/SingleTableDropSchemaMetadataValidatorTest.java

```diff
@@ -0,0 +1,94 @@
+import org.apache.shardingsphere.infra.metadata.database.schema.decorator.model.ShardingSphereSchema;
+import org.apache.shardingsphere.infra.metadata.database.schema.decorator.model.ShardingSphereTable;
+import org.apache.shardingsphere.singletable.rule.SingleTableRule;
+import org.apache.shardingsphere.sql.parser.sql.common.statement.ddl.DropSchemaStatement;
+import org.apache.shardingsphere.sql.parser.sql.common.value.identifier.IdentifierValue;
+import org.apache.shardingsphere.sql.parser.sql.dialect.statement.postgresql.ddl.PostgreSQLDropSchemaStatement;
+import org.junit.Test;
+
+import static org.junit.Assert.assertTrue;
+import static org.mockito.Mockito.RETURNS_DEEP_STUBS;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+public class SingleTableDropSchemaMetadataValidatorTest {
+    @Test(expected = ShardingSphereException.class)
+    public void assertValidateCascadeNotEmpty() {
+        SingleTableDropSchemaMetadataValidator singleTableDropSchemaMetadataValidator = new SingleTableDropSchemaMetadataValidator();
+        SingleTableRule singleTableRule = mock(SingleTableRule.class, RETURNS_DEEP_STUBS);
+        SQLStatementContext<DropSchemaStatement> sqlStatementContext = createSQLStatementContext("foo_schema", false);
+        ShardingSphereDatabase database = createMockDataBase("foo_schema");
+        singleTableDropSchemaMetadataValidator.validate(singleTableRule, sqlStatementContext, database);
+    }
+
+    @Test(expected = ShardingSphereException.class)
+    public void assertValidateSchemaNotMatch() {
+        SingleTableDropSchemaMetadataValidator singleTableDropSchemaMetadataValidator = new SingleTableDropSchemaMetadataValidator();
+        SingleTableRule singleTableRule = mock(SingleTableRule.class, RETURNS_DEEP_STUBS);
+        SQLStatementContext<DropSchemaStatement> sqlStatementContext = createSQLStatementContext("foo_schema1", true);
+        ShardingSphereDatabase database = createMockDataBase("foo_schema");
+        singleTableDropSchemaMetadataValidator.validate(singleTableRule, sqlStatementContext, database);
+    }
+
+    @Test
+    public void assertValidate() {
+        boolean isVerified = true;
+        try {
+            SingleTableDropSchemaMetadataValidator singleTableDropSchemaMetadataValidator = new SingleTableDropSchemaMetadataValidator();
+            SingleTableRule singleTableRule = mock(SingleTableRule.class, RETURNS_DEEP_STUBS);
+            SQLStatementContext<DropSchemaStatement> sqlStatementContext = createSQLStatementContext("foo_schema", true);
+            ShardingSphereDatabase database = createMockDataBase("foo_schema");
+            singleTableDropSchemaMetadataValidator.validate(singleTableRule, sqlStatementContext, database);
+        } catch (ShardingSphereException ex) {
+            isVerified = false;
+        }
+        assertTrue(isVerified);
+    }
+
+    private ShardingSphereDatabase createMockDataBase(final String schemaName) {
```

Comment:
Database is one word, please do not use `DataBase`

Reply:
typo error, had fixed